apps/arweave/src/ar_data_sync.erl

```diff
@@ -101,16 +105,216 @@ sync_random_chunk() ->
 init(#{ block_index := BI, maintain_tx_index := MaintainTXIndex }) ->
 	%% Load state from disk, if any, otherwise construct it from the
 	%% block index. Store the new state.
-	State = #state{},
-	{ok, State}.
+	gen_server:cast(?MODULE, {init, BI}),
+	{ok, #state{
+			db = ?INIT_KV("kv-data-sync"),
+			tx_root_index = ?INIT_KV("kv-tx-root-index"),
+			block_offset_index = ?INIT_KV("kv-block-offset-index"),
+			tx_index = init_tx_index(MaintainTXIndex),
+			maintain_tx_index = MaintainTXIndex}}.
```

Comment:
is any of the above intensive operations to be placed in the above gen_server cast?

Reply:
these just acquire the kv-store handle, so they should be fine. the intensive
operation we foresee will be in `load_state_from_disk/1`. that will be executed
non-blocking to the process `init`. 👍